### DISCUSSION TITLE
I'm trying to understand how the new PowerShell commands work

### DISCUSSION THREAD

**Discussion by WillPittenger** [microsoft/winget-cli#2506] [PowerShell]
I was looking at the source for the new PowerShell wrapper for winget.  But I wasn't seeing some of the code.  I couldn't find classes like `MatchResult` or enums like `CompositeSearchBehavior`.  I couldn't find their declarations even in the C++ code.

I'm also puzzled why the commands, when they return something, return `object` rather than something like the previously mentioned `MatchResult`.  I was envisioning issuing something like `(Find-WinGetPackage Microsoft.PowerToys).Install()`.  If a dedicated class were used, you could only download a bare minimum amount of data.  Downloading something like `*` (all packages) can take a while if you download all fields. Instead, you download only a few key fields.  Then you expose a member method like `Expand()` which obtains the remaining fields.

Next, what's the equivalent of `winget upgrade`?  (without the package name')  Just `Update-WinGetPackages` doesn't seem to do anything.  I don't see how to turn on interactive mode for the installer either.

Finally, what's the difference between `Get-WinGetPackage` and `Get-WinGetPackageInfo`?

**Comment by denelon** [comment]
I'll circle back with the engineering team on this. The original cmdlets were part of a hackathon and were just wrapping the .exe. The latest cmdlets were designed to work with the COM API so there might be some differences between the original intent and what was built most recently.

**Comment by WillPittenger** [comment]
I also couldn't find the source for `Get-WinGetPackageInfo`.

**Comment by WillPittenger** [comment]
Could there be a problem with a recent checkin?  Some form of versionitis?
